xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/PropertyGroupDescriptor.java

```diff
@@ -32,21 +32,43 @@
 {
     private List<String> group;
 
+    private String feature;
```

Comment:
You don't like the multi features idea ? :)

May be hard to add later without breaking the API if we have the need.

Reply:
There is actually a check done when 2 groups have a different feature assigned:
https://github.com/xwiki/xwiki-commons/blob/85199240eb14f223642c5df00e308329ee2ec35f/xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/internal/DefaultBeanDescriptor.java#L279-L284

It's just a warning here but we can even throw an exception if we want to enforce the rule (at most one feature per group).

As for multi-features, I don't quite see how it fits our need and I think it will actually bring more complexity.

Reply:
An error would be safer otherwise the `feature` which end up chosen is going to be random (since the order of the fields/methods is not guaranteed).

Reply:
I mean an exception.